### ISSUE TITLE
custom gencode gtf

### ISSUE THREAD

**Issue by hanig** [kvittingseerup/IsoformSwitchAnalyzeR#15]
Hi,
I'm using gencode.v28.annotation.gtf.gz and its accompanying gencode.v28.transcripts.fa.gz for salmon indexing ( salmon index --gencode -t gencode.v28.transcripts.fa.gz -i gencode.v28.transcripts.salmon.index). However, I couldn't import that gtf. Any ideas?

Thanks:
Step 1 of 5: Obtaining annotation...
importing GTF (this may take a while)
converting annotated CDSs
Step 2 of 5: Calculating gene expression...
  |======================================================================| 100%
Step 3 of 5: Merging gene and isoform expression...
  |==============                                                        |  20%Error in apply(isoformRepExpression[, isoIndex], 1, sd) :
  dim(X) must have a positive length
In addition: Warning message:
In importRdata(isoformCountMatrix = salmonQuant$counts, isoformRepExpression = salmonQuant$abundance,  :
  The GTF file supplied to isoformExonAnnoation does not contain the suffix '.gtf'

**Comment by opplatek** [comment]
I have received the same error when I had just one sample per condition or at least one condition had only one replicate. The "problem" in my case was in this part of the function `iso_std          = apply(   isoformRepExpression[, isoIndex], 1, sd)`, line  `apply(   isoformRepExpression[, isoIndex], 1, sd)` in `import_data.R`. I know not using without replicates is not recommended and doesn't make sense but I believe there could be just a warning.
After I "faked" each condition to have at least 2 replicates (I just copied the same results twice) I got through.
